chemicaltoolbox/autodock_vina/prepare_box/calc_vina_box_params.py

```diff
@@ -0,0 +1,59 @@
+import numpy as np
+from rdkit import Chem
+from rdkit.Chem import rdShapeHelpers
+import argparse
+
+
+def get_params(options):
+    # make sure we have a mol file by initiating rdkit mol object from input
+    mol = Chem.MolFromMolFile(options.ligand_path)
+    if not mol:
+        raise IOError
+
+    # get rdkit conformer and compute x,y,z of top and bottom corner of confounding cuboid
+    conf = mol.GetConformer()
+    params = rdShapeHelpers.ComputeConfBox(conf)
+
+    # change tuples to arrays
+    coords1 = np.array(params[0])
+    coords2 = np.array(params[1])
+
+    # get the centre of the box
+    center = np.mean((coords1, coords2), axis=0)
+
+    # calculate box dimensions
+    dims = np.abs(coords1 - coords2) + (options.tolerance,) * 3
+
+    # optionally add buffers in each direction - expansion
+    box_dims = [dims[0] + options.bufx, dims[1] + options.bufy, dims[2] + options.bufz]
+
+    with open(options.output, 'w') as f:
+        f.write(
+            """
+size_x =  {}
+size_y =  {}
+size_z =  {}
+center_x =  {}
+center_y =  {}
+center_z =  {}
+num_modes = 9999
+energy_range = 9999
+exhaustiveness = {}
+cpu = 4
+seed = 1
+            """.format(box_dims[0], box_dims[1], box_dims[2], center[0], center[1], center[2], options.exhaustiveness)
+        )
+
+
+if __name__ == "__main__":
+    parser = argparse.ArgumentParser()
+    parser.add_argument('--ligand', dest='ligand_path')
+    parser.add_argument('--config', dest='output')
+    parser.add_argument('--tol', dest='tolerance', default=0, type=float)
+    parser.add_argument('--exh', dest='exhaustiveness', default=10, type=float)
```

Comment:
we don't need both tolerance and the 3 buffer values

Reply:
Resolved by: b046d0c685b7638818660cadfb7d1c16e7d97545